### DISCUSSION TITLE
Using function inside appConfig

### DISCUSSION THREAD

**Discussion by cyrenity** [nuxt/nuxt#22509] [Questions]
I'm currently working on my Nuxt 3 app and have a query regarding calling functions from the appConfig. In my code, I have a 'Settings' button with a click action, and I'm considering whether it's appropriate to call a function or update a state variable from within the appConfig. I'd like to know if this is a recommended approach, and if so, what would be the best practice for sharing state between the appConfig and the calling component? 

```
export default defineAppConfig({
  theApp: {
    name: 'TheAwesomeApp',
    link: {
          label: 'Settings',
          icon: 'i-heroicons-cog-8-tooth',
          click: () => {
            console.log('Open settings');
            // update state or do something to trigger the effect inside the page/component 
          },
        },
     },
   })
```

**Comment by deeluxe74** [comment]
From what i understand, if you have a really small application you need to trigger the action on the child and do what you want on the parent (check [emit event](https://vuejs.org/guide/components/events.html)).

If your app have multiples component and is more complex than just a static page, you need to manage a store.
Before it was vueX, because nuxt2 is based on vue2, but now in vue3 (nuxt3), this not exist, you need to use a outside plugins for this.

The best and default recommandation from Vue3 is [pinia](https://pinia.vuejs.org/ssr/nuxt.html).
